sdks/java/core/src/main/java/org/apache/beam/sdk/testing/RunnableOnService.java

```diff
@@ -31,4 +31,4 @@
  *     }
  * </code></pre>
  */
-public interface RunnableOnService {}
+public interface RunnableOnService extends NeedsRunner {}
```

Comment:
Agreed; I think there are probably two variants:

Tests that require the capability
Tests that are used to establish the correct functioning of the capability (the "more interesting" ones of the above)

We can run the first half on only the Direct Runner, and the second half on all runners supporting the appropriate capability.

I'll write up a doc sometime this week and send it to dev